### PR TITLE
chore: Move Puck styles to a separate shadow DOM tree

### PR DESCRIPTION
**Pull request by birtles**
This is a revision of #1 that uses two shadow DOM trees instead of 2

### CODE REVIEW

File: src/puck.ts [hunk 1]
```diff
@@ -1,111 +1,151 @@
...
+        this.puck.classList.remove(className);
+      }
+    }
+
+    this.puck.classList.add(`-${theme}`);
   }
 
   unmount(): void {
-    this.puck.parentElement?.removeChild(this.puck);
+    removeContentContainer('tenten-ja-puck');
```
Comment (shirakaba) [inline src/puck.ts:128]: It’s not the end of the world, but we’re still holding a reference to `this.puck` which we’ll never use again (it’s reassigned on render()), preventing its container from being garbage-collected, so we might as well assign `this.puck = null` in here.

**Review by shirakaba — CHANGES_REQUESTED** [review]
Looks great! Let’s go with two shadow trees. Good idea about stopping mousemove events propagating from the puck. Thanks for doing the necessary refactor! Hoping to get some time to get back to the code over the weekend.

File: src/puck.ts (hunk 1 above)
Comment (birtles) [reply src/puck.ts:128]: Sure! This should be fixed now.

**Review by shirakaba — APPROVED** [review]

**Comment by shirakaba** [comment]
@birtles You’re welcome to merge at your convenience!

**Pull request merged**